apps/builder/app/builder/features/settings-panel/controls/code.tsx

```diff
@@ -11,31 +18,46 @@ export const CodeControl = ({
   onChange,
   onDelete,
 }: ControlProps<"code", "string">) => {
+  const metaOverride = {
+    ...meta,
+    control: "text" as const,
+  };
+  const localValue = useLocalValue(prop?.value ?? "", (value) => {
+    // sanitize html before saving
+    // this is basically what browser does when innerHTML is set
+    // but isolated within temporary element
+    // so the result is correct markup
+    const div = document.createElement("div");
+    div.innerHTML = value;
```

Comment:
is there risk of executing scripts etc? I don't remember if it does ...

Reply:
innerHTML doesn't execute scripts